src/test/java/streams/part1/exercise/Exercise2.java

```diff
@@ -0,0 +1,116 @@
+package streams.part1.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+class Exercise2 {
+
+    @Test
+    void calcAverageAgeOfEmployees() {
+        List<Employee> employees = getEmployees();
+
+        Double expected = employees.stream()
+                .mapToDouble(employee -> employee.getPerson().getAge())
+                .average().orElse(-1);
+
+        assertThat(expected, Matchers.closeTo(33.66, 0.1));
+    }
+
+    @Test
+    void findPersonWithLongestFullName() {
+        List<Employee> employees = getEmployees();
+
+        Person expected = employees.stream()
+                .map(Employee::getPerson)
+                .max(Comparator.comparingInt(p -> p.getFullName().length()))
+                .orElse(new Person());
```

Comment:
I doubt that returning a Person with default field values is a good idea. In this case you'd need to check if returned person's name is not 'default'. I would throw an exception here.

Reply:
Done